src/typing/fields.ml

```diff
@@ -330,6 +330,7 @@ let rec using_field ctx mode e i p =
 
 (* Resolves field [i] on typed expression [e] using the given [mode]. *)
 let rec type_field cfg ctx e i p mode =
+	let pfield = if (e.epos == p) then p else {p with pmin = e.epos.pmax + 1} in
```

Comment:
I don't think you want `==` here because physical equality of positions doesn't mean much. This should be `=` instead.

Also, did you check this with something like `expr           .field`? This can happen with fluent interface syntax.

Reply:
![img](https://i.imgur.com/UODt0ih.png)

Reply:
Thanks, sorry I forgot about this PR

Reply:
I don't understand how `{p with pmin = e.epos.pmax + 1}` would give the correct position here...

Reply:
It's like transforming an `EField(e:Expr, field:String)` into an expr for `field`: `e` is the expression until the dot before `field` (while `p` here is the position of the whole expression).

Reply:
Yeah but that's my point, wouldn't pmax + 1 being just after the `)`?

Reply:
Oh you mean in the screenshot.. beats me actually xD

Reply:
In another place I use `{ p with pmin = p.pmax - String.length s }` where `s` is the field name. That should give the correct result in all cases.

Reply:
done